ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleBannerAdapter.java

```diff
@@ -19,38 +21,70 @@
 import com.vungle.warren.error.VungleException;
 import java.lang.ref.WeakReference;
 
-class VungleBannerAdapter {
+class VungleBannerAdapter implements PlayAdCallback {
 
   private static final String TAG = VungleBannerAdapter.class.getSimpleName();
 
-  /** Vungle banner placement ID. */
-  @NonNull private String mPlacementId;
+  /**
+   * Vungle banner placement ID.
+   */
+  @NonNull
+  private String mPlacementId;
 
-  /** Unique Vungle banner request ID. */
-  @Nullable private String mUniquePubRequestId;
+  /**
+   * Unique Vungle banner request ID.
+   */
+  @Nullable
+  private String mUniquePubRequestId;
+
+  /**
+   * Mediation Banner Adapter instance to receive callbacks.
+   */
+  private MediationBannerAdapter bannerAdapter;
 
-  /** Vungle listener class to forward to the adapter. */
-  @NonNull private WeakReference<VungleListener> mVungleListener = new WeakReference<>(null);
+  /**
+   * Vungle listener class to forward to the adapter.
+   */
+  private MediationBannerListener bannerListener;
```

Comment:
@palrahul @jiancchen 
To avoid the leak of VungleInterstitialAdapter (and VungleBannerAdapter it holds), we let VungleInterstitialAdapter GCed and destroy VungleBannerAdapter with calling VungleBanner#finishAd unblocking that placement.
That's why we can't MediationBannerListener because it holds a reference to VungleInterstitialAdapter preventing garbage collecting

Reply:
@palrahul @jiancchen 
why we need a weak reference to MediationBannerListener (VungleInterstitialAdapter has already leaked, but can't be GCted):
![image](https://user-images.githubusercontent.com/43134123/94709354-4bdc6700-034e-11eb-82e2-962e3d7b3b1c.png)